webserver/views/api.py

```diff
@@ -248,20 +250,30 @@ def _messybrainz_lookup(listens):
     return augmented_listens
 
 
-def _validate_listen(listen):
+def _validate_listen(listen, listen_type):
```

Comment:
I would prefer to have a numerical constant, rather than a string here. It is much faster to use an int that a string for this. Define and use the LISTEN_TYPE_SINGLE, LISTEN_TYPE_IMPORT and LISTEN_TYPE_NOW_PLAYING.